### PR TITLE
Refactor entrypoint script

### PR DESCRIPTION
**Pull request by Ulexus**
The entrypoint script has gotten a bit ragged; this is an attempt to refactor it and make it a bit more flexible, with less repetition.

In particular, I have broken out the "static" and "kv" backends, which both adds separation and leaves open the option of others providing their own configuration backends. More thought and work is probably in order.

### CODE REVIEW

File: daemon/entrypoint.sh [hunk 1]
```diff
@@ -304,47 +123,54 @@ fi
...
+         osd_directory
+         ;;
+      disk)
+         osd_disk
+         ;;
+      activate)
+         osd_activate
+         ;;
+      *)
+         if [ -n "$(find /var/lib/ceph/osd -prine -empty)" ]; then
```
Comment (hunter) [inline daemon/entrypoint.sh]: Should the arg here but `-prune`?

**Comment by hunter** [comment]
So much cleaner! A big +1 from me :)

File: daemon/entrypoint.sh (hunk 1 above)
Comment (Ulexus) [inline daemon/entrypoint.sh]: Thanks!

**Comment by leseb** [comment]
I like this approach too, I believe the same path should be followed for internal daemons.
So could also stripe this to monitor, osd, mds, radosgw and restapi.
This can be part of another PR though.

@Ulexus stupid question maybe, but did you get the chance to test this out? If so and if everything is fine I'm good for merging this work. If it's not the case I can find some time to test this.

**Comment by Ulexus** [comment]
@leseb No, it's only rudimentarily tested.  I'll see if I can throw up a cluster and test it more throughly; I'm on holiday for the next couple of weeks, so it might not happen quickly.

**Comment by leseb** [comment]
@Ulexus Ok thanks for letting me know, I'll then try to find some time to test this and eventually merge it.
Enjoy your holiday!

File: daemon/entrypoint.sh [hunk 2]
```diff
@@ -462,34 +277,34 @@ elif [[ "$CEPH_DAEMON" = "OSD_CEPH_DISK_ACTIVATE" ]]; then
...
+       KEYRING_OPT="--keyring /etc/ceph/${CLUSTER}.client.admin.keyring"
+    elif [ -e /var/lib/ceph/bootstrap-mds/ceph.keyring ]; then
+       KEYRING_OPT="--keyring /var/lib/ceph/bootstrap-mds/ceph.keyring"
+    else
+      echo "ERROR- Failed to bootstrap MDS: could not find admin or bootstrap-mds keyring.  You can extract it from your current monitor by running 'ceph auth get client.bootstrap-mds -o /var/lib/ceph/bootstrap-mds/ceph.keyring'"
       exit 1
     fi
 
-    timeout 10 ceph --cluster ${CLUSTER} --name client.bootstrap-mds --keyring /var/lib/ceph/bootstrap-mds/ceph.keyring health || exit 1
+    timeout 10 ceph --cluster ${CLUSTER} --name client.bootstrap-mds $KEYRING_OPT || exit 1
```
Comment (leseb) [inline daemon/entrypoint.sh]: You forgot the "health", the command is broken.

Comment (Ulexus) [inline daemon/entrypoint.sh]: _That's_ why that command didn't make sense!  Thanks

**Comment by leseb** [comment]
I tested all the daemons, everything works with the last mds fix, merging this now.

**Comment by Ulexus** [comment]
Thanks @leseb

**Pull request merged**